File: mobile/lib/features/leads/domain/entities/lead.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'lead.freezed.dart';
part 'lead.g.dart';

@freezed
abstract class Lead with _$Lead {
  const factory Lead({
    required String id,
    required String title,
    required String name,
    @JsonKey(name: 'company_name') String? company,
    String? email,
    String? phone,
    required String source,
    required String status,
    @JsonKey(name: 'customer_id') String? customerId,
    @JsonKey(name: 'estimated_value') double? estimatedValue,
    @JsonKey(name: 'potential_products') List<String>? potentialProducts,
    String? notes,
    @JsonKey(name: 'converted_at') DateTime? convertedAt,
    double? latitude,
    double? longitude,
  }) = _Lead;

  factory Lead.fromJson(Map<String, dynamic> json) => _$LeadFromJson(json);
}

File: backend/internal/domain/models/lead.go
package models

import (
	"crm_wowin_backend/pkg/utils"

	"github.com/google/uuid"
)

type LeadStatus string

const (
	LeadStatusNew         LeadStatus = "new"
	LeadStatusContacted   LeadStatus = "contacted"
	LeadStatusQualified   LeadStatus = "qualified"
	LeadStatusUnqualified LeadStatus = "unqualified"
)

type LeadSource string

const (
	LeadSourceSurvey      LeadSource = "survey"
	LeadSourceReferral    LeadSource = "referral"
	LeadSourceColdCall    LeadSource = "cold_call"
	LeadSourceSocialMedia LeadSource = "social_media"
	LeadSourceWebsite     LeadSource = "website"
	LeadSourceEvent       LeadSource = "event"
	LeadSourceOther       LeadSource = "other"
)

// Lead represents a raw prospect before standardizing to Customer/Deal
type Lead struct {
	ID             uuid.UUID   `json:"id"`
	Title          string      `json:"title"`
	Name           string      `json:"name"`
	Company        *string     `json:"company,omitempty"`
	Email          *string     `json:"email,omitempty"`
	Phone          *string     `json:"phone,omitempty"`
	Source         LeadSource  `json:"source"`
	Status         LeadStatus  `json:"status"`
	AssignedTo     *uuid.UUID  `json:"sales_id,omitempty"`
	CustomerID        *uuid.UUID  `json:"customer_id,omitempty"` // populated if converted
	EstimatedValue    *float64    `json:"estimated_value,omitempty"`
	PotentialProducts []string    `json:"potential_products,omitempty"`
	Notes             *string     `json:"notes,omitempty"`
	ConvertedAt       *utils.FlexTime `json:"converted_at,omitempty"`
	Address           *string     `json:"address,omitempty"`
	Latitude          *float64    `json:"latitude,omitempty"`
	Longitude         *float64    `json:"longitude,omitempty"`
	CreatedBy      *uuid.UUID  `json:"created_by,omitempty"`
	CreatedAt      utils.FlexTime   `json:"created_at"`
	UpdatedAt      utils.FlexTime   `json:"updated_at"`
}
